.github/workflows/CI.yml

```diff
@@ -100,9 +100,25 @@ jobs:
     name: release
     needs: [build, clippy, test, test-script, fmt]
     if: github.event_name == 'workflow_dispatch' && github.ref == 'refs/heads/main'
+    permissions:
+      contents: write  # for actions/checkout to fetch code and for semantic-release to push commits, release releases and tags
+      issues: write # for semantic-release to comment on and close issues
+      pull-requests: write # for semantic-release to comment on and close pull requests
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@stable
-      - run: cargo publish
+      - name: Semantic Release
+        uses: cycjimmy/semantic-release-action@v4
```

Comment:
```suggestion
        uses: Wulf/semantic-release-action@v4
```

Reply:
could you explain this change? do you want to be sure a specific version is used instead of a "third-party user"?

Reply:
The intent was to avoid a potential supply chain attack

Reply:
sure, i usually dont do this because i would like it to stay "updated" (for github actions, without having to do more than update the version tag)

if we want to avoid a attack, just using a commit hash would also do instead of a whole fork (see [mongoose's workflows](https://github.com/Automattic/mongoose/blob/master/.github/workflows/test.yml))

Reply:
that works! let's pin it to a SHA

Reply:
done with 7e20dbaf7104b8c0a08fddfa1b1d45d7c9d67595